src/Poseidon/Package.hs

```diff
@@ -530,6 +526,15 @@ getJointGenotypeData showAllWarnings intersect pacs maybeSnpFile = do
         (Just snp, Nothing) -> unless intersect $ yield (snp, V.replicate n Missing)
         _ ->  return ()
 
+getJointJanno :: [PoseidonPackage] -> [JannoRow]
+getJointJanno = concatMap posPacJanno
+
+getJointIndividualInfo :: [PoseidonPackage] -> [IndividualInfo]
+getJointIndividualInfo packages = do
+    pac <- packages
+    jannoRow <- posPacJanno pac
+    return $ IndividualInfo (jPoseidonID jannoRow) ((getJannoList . jGroupName) jannoRow) (posPacTitle pac)
```

Comment:
I think multi-line list-comprehension (aka do-notation for lists) are one of the killer features in Haskell. I have actually started a blog-post about it ("List comprehensions on steroids"), because the only way to come close to that in python or R would be to define a function and then map it over the list, which is arguably less readable. Furthermore, this is even a nested-for-loop, with an external and internal loop, and the do-notation automatically collapses them to a one-dimensional output. That's not even possible in python or R just like that... I don't think there is a better or more succinct way of writing that. I'm a huge fan of multi-line list comprehensions in Haskell!

BUT: As one of only two Haskell-developers in this project you have a big say here, and I'm very open to suggestions to make the code more readable :-)

Reply:
Alright - let's keep it then. List comprehensions exist in Python and R, btw.. No idea if they are this elegant, though.